src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -189,9 +190,11 @@ class spell_gen_animal_blood : public AuraScript
 
     void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
     {
+        if (GetTargetApplication()->GetRemoveMode() == AURA_REMOVE_BY_EXPIRE)
```

Comment:
expire isnt the only reason for this aura to be removed outside of water (interrupt flag)
immunity (divine shield?)
death

Reply:
Although the spell IsDeathPersistent, it seems to be broken and gets removed, but code triggers on logout anyway

Reply:
Not sure what to do here, to match the exact desired behavior, the calls to `RemoveAurasWithInterruptFlags(AURA_INTERRUPT_FLAG_NOT_ABOVEWATER)` needs to be changed.

From `Unit::ProcessTerrainStatusUpdate` (triggers by stepping on water, shallow or not)
To `Player::SetInWater(bool apply)` (triggers only when "swimming") <- This is what seems blizzlike

Reply:
how is the status of this ?

Reply:
Decided to just merge the missing script, the remaining issue is due to generic changes that I'm not confident enough to touch, like #23091 and https://github.com/TrinityCore/TrinityCore/pull/23088#discussion_r262303442